framework/src/meshgenerators/Poly2TriMeshGenerator.C

```diff
@@ -0,0 +1,409 @@
+    {
+      // We'll probably need to make a new ID, then
+      _output_subdomain_id = MooseMeshUtils::getNextFreeSubdomainID(*mesh);
+
+      // But check the hole meshes for our output subdomain name too
+      for (auto hole_ptr : _hole_ptrs)
+      {
+        auto possible_sbdid = MooseMeshUtils::getSubdomainID(output_subdomain_name, **hole_ptr);
+        // Huh, it was in one of them
+        if (possible_sbdid != Elem::invalid_subdomain_id)
+        {
+          _output_subdomain_id = possible_sbdid;
+          break;
+        }
+        _output_subdomain_id =
+            std::max(_output_subdomain_id, MooseMeshUtils::getNextFreeSubdomainID(**hole_ptr));
+      }
+
+      mesh->subdomain_name(_output_subdomain_id) = output_subdomain_name;
+    }
+  }
+
+  if (_smooth_tri || _output_subdomain_id)
+    for (auto elem : mesh->element_ptr_range())
+    {
+      mooseAssert(elem->type() == TRI3, "Unexpected non-Tri3 found in triangulation");
+
+      elem->subdomain_id() = _output_subdomain_id;
+
+      // I do not trust Laplacian mesh smoothing not to invert
+      // elements near reentrant corners.  Eventually we'll add better
+      // smoothing options, but even those might have failure cases.
+      // Better to always do extra tests here than to ever let users
+      // try to run on a degenerate mesh.
+      if (_smooth_tri)
+      {
+        auto cross_prod = (elem->point(1) - elem->point(0)).cross(elem->point(2) - elem->point(0));
+
+        if (cross_prod(2) <= 0)
+          mooseError("Inverted element found in triangulation.\n"
+                     "Laplacian smoothing can create these at reentrant corners; disable it?");
+      }
+    }
+
+  // Assign new subdomain name, if provided
+  if (isParamValid("output_subdomain_name"))
+    mesh->subdomain_name(_output_subdomain_id) = getParam<SubdomainName>("output_subdomain_name");
+
+  const bool use_binary_search = (_algorithm == "BINARY");
+
+  // The hole meshes are specified by the user, so they could have any
+  // BCID or no BCID or any combination of BCIDs on their outer
+  // boundary, so we'll have to set our own BCID to use for stitching
+  // there.  We'll need to check all the holes for used BCIDs, if we
+  // want to pick a new ID on hole N that doesn't conflict with any
+  // IDs on hole M < N (or with the IDs on the new triangulation)
+
+  // The new triangulation by default assigns BCID i+1 to hole i ...
+  // but we can't even use this for mesh stitching, because we can't
+  // be sure it isn't also already in use on the hole's mesh and so we
+  // won't be able to safely clear it afterwards.
+  const boundary_id_type end_bcid = _hole_ptrs.size() + 1;
+  boundary_id_type new_hole_bcid = end_bcid;
+
+  // We might be overriding the default bcid numbers.  We have to be
+  // careful about how we renumber, though.  We pick unused temporary
+  // numbers because e.g. "0->2, 2->0" is impossible to do
+  // sequentially, but "0->N, 2->N+2, N->2, N+2->0" works.
```

Comment:
This is one of the reasons why I'm not a fan of the extra parameters. The rename boundary generator does support swapping by doing this temp changing

Reply:
It is a trade-off of minimizing the number of lines in inputs and the number of lines in codes. Another downside of using rename boundary generator is that users have to remember what side sets this generator added, which can be difficult with auto id-assigment.

Reply:
It's a maintenance burden.

Reply:
Yes. But worthwhile if users feel the syntax is more friendly. I think it needs to be clear in an input about the linkage among parts of the input, i.e. an object name required from users should be created by users in the same input. This means having a parameter for users to set the boundary name in this generator anyway.

Reply:
I would just add a subgenerator to rely on the rename generator to do the same thing.

Reply:
It's not "the" rename boundary generator, if we want to use it here, though.  It's "a rename boundary generator for every input mesh, plus one more rename boundary generator for the output mesh" ... and because of that, that workflow can't do automatic temp ids like the code here does, because a temporary id has to be independent not only of other ids in the current mesh but of other ids in other meshes that the input rename generators don't know exist.

The subgenerator idea is worth looking into, though.  Might save a couple lines of code?

Reply:
> Might save a couple lines of code?

Nope.  Discussion in https://github.com/idaholab/moose/pull/21800#discussion_r971013551